File: app/src/main/java/com/github/weatherapp/ui/base/BaseMvpPresenter.java
package com.github.weatherapp.ui.base;

/*
    Basic presenter functionality
 */
public class BaseMvpPresenter<V extends BaseMvpView> {
    private V view;

    public void attach(V view) {
        this.view = view;
    }

    public void detach() {
        view = null;
    }

    protected V getView() {
        return view;
    }

    protected boolean isViewAttached() {
        return view != null;
    }
}

File: app-kotlin/src/main/kotlin/com/github/weatherapp/ui/base/BaseMvpPresenter.kt
package com.github.weatherapp.ui.base

/*
    Basic presenter functionality
 */
open class BaseMvpPresenter<V : BaseMvpView> {
    protected var view: V? = null
        private set

    fun attach(view: V) {
        this.view = view
    }

    fun detach() {
        view = null
    }
}
